backend/tests/unit/core/diff/test_diff_and_merge.py

```diff
@@ -117,6 +147,113 @@ async def test_diff_and_merge_schema_with_default_values(
         assert "nickname" not in attribute_names
         await verify_no_duplicate_paths(db=db)
 
+    @pytest.mark.parametrize(
+        "base_action,diff_action,selection,expect_deleted",
+        [
+            # DELETE on base + UPDATE on diff: selecting BASE_BRANCH keeps the node deleted
+            ("delete", "update", ConflictSelection.BASE_BRANCH, True),
+            # UPDATE on base + DELETE on diff: selecting BASE_BRANCH keeps the node with base value
+            ("update", "delete", ConflictSelection.BASE_BRANCH, False),
+            # Note: DIFF_BRANCH selection for node-level conflicts involving deletion is not tested here
+            # because un-deleting an object on the default branch is not supported
+        ],
+    )
```

Comment:
Just a reminder of that pattern we use with dataclasses for a testcase, with something like that the comments could also go into a description of each test case. This test is fairly contained as it is to could be fine to use this format here, but in general I think the alternative is easier to read.